### PR TITLE
Reserve bounty improvements

### PR DESCRIPTION
**Pull request by vince0656**
##### Description

Inprovements to the reserve bounty functionality including a new 'reserved' state for bounties so that they don't show up on the issue explorer when users are searching for 'open' bounties

##### Refers/Fixes

https://github.com/gitcoinco/web/issues/3544

##### Testing

Appropriate unit tests have been added

##### Video demonstration

https://drive.google.com/open?id=1-WZrPWReWAC2h6-u54FmU7phD9GX9iCA

### CODE REVIEW

**Comment by thelostone-mc** [comment]
@vince0656 nice ! mind opening a draft PR going forward so that it's extra clear that the PR is not ready for review ?

**Comment by vince0656** [comment]
> @vince0656 nice ! mind opening a draft PR going forward so that it's extra clear that the PR is not ready for review ?

Sure! I've updated the title. Is that sufficient or is there anything else I need to do to make it clear it's a draft PR?

**Comment by danlipert** [comment]
@vince0656 Hey Vincent - when you create a new PR, theres an option to create it as a normal PR, or a draft PR. The difference is that it doesn't ping us for reviews until you set it ready for review if its a draft PR 👍

**Comment by thelostone-mc** [comment]
sweet looks like we making progress ^_^

**Comment by vince0656** [comment]
@danlipert @thelostone-mc @PixelantDesign All good to go 👍

File: app/dashboard/models.py [hunk 1]
```diff
@@ -1151,15 +1156,59 @@ def can_remarket(self):
...
+            if self.reserved_for_user_expiration is None:
+                return 'indefinitely'
+
+            if self.reserved_for_user_from == self.reserved_for_user_expiration:
+                return ''
+
+            delta = self.reserved_for_user_expiration - self.reserved_for_user_from
+            days = delta.days
+
+            if days > 0:
```
Comment (danlipert) [inline app/dashboard/models.py]: Is there a simpler way we can humanize the timedelta?  https://anandnalya.com/2009/05/humanizing-the-time-difference-in-django/

Comment (vince0656) [reply app/dashboard/models.py]: Hey Dan. I'm not sure what you mean? I've had a look at the link you sent and we would still need to do calcs for hours. Additionally, the humanised timedelta outputs seem different compared to the current requirement of allowing a funder to reserve for '3 days', '1 week' or 'Keep reserved'. I tried to make the code as scalable as possible and struggled to find easier ways of doing this. It's a shame the timedelta object doesn't have an hours getter attribute or even a weeks one which is strange as you can construct a timedelta with weeks, days, hours, minutes etc.

Comment (vince0656) [reply app/dashboard/models.py]: @danlipert

**Comment by vince0656** [comment]
I’m excited too! Any thoughts @danlipert?

File: app/dashboard/models.py (hunk 1 above)
Comment (vince0656) [reply app/dashboard/models.py]: @danlipert any thoughts on my comment?

File: app/assets/v2/css/submit_bounty.css [hunk 2]
```diff
@@ -227,3 +227,13 @@ input:read-only {
 .letter-spacing {
   letter-spacing: 0.1em;
 }
+
+.releaseAfterFormGroup {
```
Comment (octavioamu) [inline app/assets/v2/css/submit_bounty.css]: Please lets follow the class name convention not camelcase

Comment (octavioamu) [reply app/assets/v2/css/submit_bounty.css]: After that this will be good for merge @vince0656

Comment (vince0656) [reply app/assets/v2/css/submit_bounty.css]: ok no problem, I'll make that change and push it up

**Pull request merged**